infrastructure/src/main/java/org/corfudb/infrastructure/configuration/ServerConfiguration.java

```diff
@@ -0,0 +1,722 @@
+    private static final String LOG_SIZE_QUOTA = "log-size-quota-percentage";
+
+
+    // Sequencer parameters
+    private static final String SEQUENCER_CACHE_SIZE = "sequencer-cache-size";
+
+    // Management parameters
+    private static final String STATE_TRANSFER_BATCH_SIZE = "state-transfer-batch-size";
+    private static final String NUM_MANAGEMENT_SERVER_THREADS = "management-server-threads";
+
+    //Added parameters
+    private static final String AUTO_COMMIT = "auto-commit";
+    private static final String MAX_REPLICATION_DATA_MESSAGE_SIZE = "max-replication-data-message-size";
+    private static final String COMPACT_RATE = "compact-rate";
+    private static final String PLUGIN_CONFIG_FILE_PATH = "plugin-config-file-path";
+    private static final String ENABLE_METRICS = "metrics-enabled";
+    private static final String SNAPSHOT_BATCH_SIZE = "snapshot-batch";
+    private static final String LOCK_LEASE_DURATION = "lock-lease";
+    private static final String THREAD_PREFIX = "thread-prefix";
+    private static final String BIND_TO_ALL_INTERFACES = "bindToAllInterfaces";
+
+    //Default Configuration Values
+    private static final String DEFAULT_LOG_UNIT_CACHE_RATIO = "0.5";
+    private static final String DEFAULT_LOG_LEVEL = "INFO";
+    private static final String DEFAULT_COMPACT_RATE = "60";
+    private static final String DEFAULT_BASE_SERVER_THREADS = "1";
+    private static final String DEFAULT_LOG_SIZE_QUOTA = "100.0";
+    private static final String DEFAULT_LOG_UNIT_WORKER_THREADS = "4";
+    private static final String DEFAULT_MANAGEMENT_SERVER_THREADS = "4";
+    private static final String DEFAULT_IO_THREADS = "4";
+    private static final String DEFAULT_SEQUENCER_CACHE_SIZE = "250000";
+    private static final String DEFAULT_STATE_TRANSFER_BATCH_SIZE = "100";
+    private static final String DEFAULT_CHANNEL_IMPLEMENTATION_TYPE = "nio";
+    private static final String DEFAULT_HANDSHAKE_TIMEOUT = "10";
+    private static final String DEFAULT_CLUSTER_ID = "auto";
+    private static final String DEFAULT_TLS_CIPHERS = "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256";
+    private static final String DEFAULT_TLS_PROTOCOLS = "TLSv1.1,TLSv1.2";
+    private static final String DEFAULT_THREAD_PREFIX = "";
+    private static final String DEFAULT_METADATA_RETENTION = "1000";
+
+    // The underlying map of PropertiesConfiguration can't be used to store an EventLoopGroup,
+    // so a separate map is needed. This shouldn't be here, but the Unit Tests rely on
+    // these event loops to be here
+    private final Map<String, EventLoopGroup> testEventLoops = new HashMap<>();
+
+
+
+
+    public static ServerConfiguration getServerConfigFromFile(String configFilePath) {
+        FileInputStream configFileStream;
+        Properties configProperties = new Properties();
+
+        try {
+            configFileStream = new FileInputStream(configFilePath);
```

Comment:
This can be wrapped in a try-with-resources block, or alternatively,  it can be closed explicitly in a finally clause.